File: Csharp/RT_CS/RT_CS/Lights/PointLight.cs
using RT_CS.utility;
using System;
using System.Collections.Generic;
using System.Text;

namespace RT_CS.Lights
{
    public class PointLight : Light
    {
        protected Vector3 origin;

        public PointLight(int lid, Vector3 d, Vector3 c,Vector3 o) : base(lid,d,c)
        {
            origin = o;
        }

        public override Vector3 GetDirection(Vector3 point)
        {
            return (point - origin).Normalize();
        }
    }
}

File: cpp/RT_CPP/RT_CPP/point_light.cpp
#include "point_light.h"
#include "operators.h"

using namespace utility;
using namespace lights;

point_light::point_light(int id, vector3 color, vector3 direction, vector3 o) : light(id, color, direction),
origin(o)
{

}

vector3 point_light::get_direction(vector3 coordinate) const
{
	return (coordinate - origin).normalize();
}

float point_light::light_point_distance(vector3 point)const
{
	return(point - origin).magnitude_sq();
}
